torch/csrc/Module.cpp

```diff
@@ -165,6 +166,7 @@ bool THPModule_isTensor(PyObject *obj)
 PyObject * THPModule_setDefaultTensorType(PyObject *_unused, PyObject *type)
 {
   THPDefaultTensorClass = type;
+  THPDefaultATenType = &torch::getATenType((PyTypeObject*)type);
```

Comment:
do we have ATen types for all supported PyTorch types?  maybe not SparseCudaHalf? (although I can't seem to set that as the default anyway).  This looks okay then.

Reply:
I don't think we have that outside of ATen either. Anyways, if you try setting that as the default tensor type you're going to be very, very sorry